### PR TITLE
Error Checks and Dead Code Removal

### PR DESCRIPTION
**Pull request by daniellockard**
I'm sorry this is so huge - I know it might take a bit to review. I tried to follow the code style of what was around these calls to either return the error, or print it.  Part of the issue is that the `Log` calls all can return an Error, and there are cases where we're using the logger to log an error, in that case I think it's best to Println.

### CODE REVIEW

File: cmd/mdmctl/mdmcert.download.go [hunk 1]
```diff
@@ -213,7 +219,10 @@ func (cmd *mdmcertDownloadCommand) Run(args []string) error {
 		if err != nil {
 			return errors.Wrap(err, "reading PKI private key")
 		}
-		ioutil.WriteFile("/tmp/fubar.p7", pkcsBytes, 0666)
+		err = ioutil.WriteFile("/tmp/fubar.p7", pkcsBytes, 0666)
```
Comment (jessepeterson) [inline cmd/mdmctl/mdmcert.download.go]: This looks like never-removed debugging code. Not sure why we'd need to write this out.

Comment (daniellockard) [reply cmd/mdmctl/mdmcert.download.go]: I doubt we do :), especially with a name like "fubar"

File: cmd/mdmctl/mdmcert.go [hunk 2]
```diff
@@ -178,7 +184,10 @@ func (cmd *mdmcertCommand) runPush(args []string) error {
 	}
 
 	if err := os.MkdirAll(filepath.Dir(*flCSRPath), 0755); err != nil {
-		errors.Wrapf(err, "create directory %s", filepath.Dir(*flCSRPath))
+		innerErr := errors.Wrapf(err, "create directory %s", filepath.Dir(*flCSRPath))
+		if innerErr != nil {
+			return innerErr
```
Comment (jessepeterson) [inline cmd/mdmctl/mdmcert.go]: any reason for not just returning the Wrapf here?

File: cmd/mdmctl/apply.go [hunk 3]
```diff
@@ -42,7 +42,10 @@ func (cmd *applyCommand) setup() error {
 
 func (cmd *applyCommand) Run(args []string) error {
 	if len(args) < 1 {
-		cmd.Usage()
+		err := cmd.Usage()
```
Comment (groob) [inline cmd/mdmctl/apply.go]: would prefer to see single returns on the same line as the if
` if err := (); err != nil ...`

File: cmd/mdmctl/mdmcert.download.go [hunk 4]
```diff
@@ -96,13 +96,19 @@ func (cmd *mdmcertDownloadCommand) Run(args []string) error {
...
+		if innerErr != nil {
+			return errors.Wrap(innerErr, "calling usage")
+		}
+		return errors.Wrap(err, "failed parsing")
 	}
 
 	// neither flag was used
 	if !*flNew && *flDecrypt == "" {
-		cmd.Usage()
+		innerErr := cmd.Usage()
```
Comment (groob) [inline cmd/mdmctl/mdmcert.download.go]: if err := cmd.Usage(); err != nil {

File: mdm/appmanifest/appmanifest_test.go [hunk 5]
```diff
@@ -39,7 +39,10 @@ func benchmarkSize(b *testing.B, size int) {
 	r := bytes.NewReader(buf)
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		calculateMD5s(r, DefaultMD5Size)
+		_, err := calculateMD5s(r, DefaultMD5Size)
+		if err != nil {
+			return
```
Comment (groob) [inline mdm/appmanifest/appmanifest_test.go]: b.Fatal(err)

File: pkg/httputil/httputil.go [hunk 6]
```diff
@@ -69,7 +69,10 @@ func ErrorEncoder(_ context.Context, err error, w http.ResponseWriter) {
 	}
 	w.WriteHeader(code)
 
-	enc.Encode(errMap)
+	err = enc.Encode(errMap)
+	if err != nil {
+		fmt.Println(errors.Wrap(err, "encoding errMap"))
```
Comment (groob) [inline pkg/httputil/httputil.go]: need to either use the logger or write a 500. I think not checking the error here is preferable. Failing to encode just means we can't write back to the client for some reason.

File: platform/apns/worker.go [hunk 7]
```diff
@@ -45,10 +46,13 @@ func (w *Worker) Run(ctx context.Context) error {
 			err = w.updatePushInfoFromTokenUpdate(ctx, event.Message)
 		}
 		if err != nil {
-			level.Info(w.logger).Log(
+			innerErr := level.Info(w.logger).Log(
```
Comment (groob) [inline platform/apns/worker.go]: logger should not be checked for error. the error is only used for audit logging.
FYI, `fmt.Print` returns an error too. 
https://golang.org/pkg/fmt/#Println

Comment (daniellockard) [reply platform/apns/worker.go]: Ah, you're not wrong. I'll rip out all the logger error checking.

**Pull request closed without merging**